qcelemental/datum.py

```diff
@@ -26,18 +26,29 @@ class Datum(BaseModel):
         Literature citation or definition DOI link.
     glossary : str, optional
         Extended description or definition.
+    numeric : bool, optional
+        Whether `data` is numeric. Pass `True` to disable validating `data` as float/Decimal/np.ndarray.
 
     """
 
+    numeric: bool
     label: str
     units: str
     data: Any
     comment: str = ''
     doi: Optional[str] = None
     glossary: str = ''
 
-    def __init__(self, label, units, data, *, comment=None, doi=None, glossary=None):
-        kwargs = {'label': label, 'units': units, 'data': data}
+    class Config:
+        extra = "forbid"
+        allow_mutation = False
+        json_encoders = {
+            np.ndarray: lambda v: v.flatten().tolist(),
+            complex: lambda v: (v.real, v.imag),
```

Comment:
Looks like no built in Pydantic support for complex numbers, we might consider adding it there.

Reply:
that'd be great. especially as this is going to cause trouble for reconstituting (complex or tuple?).

I'd like to get this is as a stopgap for now so I can clear out some PRs.